netbox/utilities/middleware.py

```diff
@@ -21,9 +21,7 @@ def __call__(self, request):
             # Redirect unauthenticated requests to the login page. API requests are exempt from redirection as the API
             # performs its own authentication. Also metrics can be read without login.
             api_path = reverse('api-root')
-            if (not (request.path_info.startswith(api_path) or
-                     request.path_info.startswith('/metrics')) and
-                    request.path_info != settings.LOGIN_URL):
+            if not request.path_info.startswith(api_path, '/metrics') and request.path_info != settings.LOGIN_URL:
```

Comment:
You have to pass a tuple:

`(api_path, '/metrics')` instead of just `api_path, '/metrics'`

Reply:
Just saw that I missed the additional brackets for the tuple… Will be pushed in a few seconds.